cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -65,12 +66,20 @@ func newDevicePlugin(sysfsDir, devfsDir string, sharedDevNum int) *devicePlugin
 }
 
 func (dp *devicePlugin) Scan(notifier dpapi.Notifier) error {
+	var previouslyFound int = -1
+
 	for {
 		devTree, err := dp.scan()
 		if err != nil {
 			return err
 		}
 
+		found := len(devTree)
```

Comment:
oh, and I meant to note, I'm not convinced directly accessing the `devTree` here is a good thing - it feels like breaking the API contract and directly accessing what should maybe be an opaque object.... maybe we need a `.len()` accessor etc.